enzy_htp/preparation/protonate.py

```diff
@@ -304,13 +310,41 @@ def _fix_pybel_output(pdb_path: str, out_path: str, ref_name_path: str = None) -
     use {ref_name_path} to maximumly keep original atom names and residue name (if provided)
     * this fix is based on all newly added atoms from pybel will be under original ones
     * i.e.: assume the original atoms is a subset of the final atoms and in its original order
+
+    Args:
+        pdb_path: Protonated PDB filepath with messed up atom names and residue names.
+        out_path: Fixed PDB filepath.
+        ref_name_path: Reference PDB filepath for atom names and residue name.
+
+    Returns:
+        None.
     """
     if ref_name_path is not None:
         ref_atom_names = []
         ref_ligand = PandasPdb()
         ref_ligand.read_pdb(ref_name_path)
         ref_ligand_df: pd.DataFrame = pd.concat((ref_ligand.df["ATOM"], ref_ligand.df["HETATM"]), ignore_index=True)
         ref_ligand_df.sort_values("line_idx", inplace=True)  # make sure lines are aligned
+
+        # (Zhong) Find the first hydrogen atom in the element column
+        # and throws a ValueError if it is followed by any heavy atom anywhere.
+        if 'element_symbol' not in ref_ligand_df.columns:
+            exception_msg = f'{ref_name_path}: Element Symbol field does not exist in the PDB file passed via `ref_name_path`'
+            _LOGGER.exception(exception_msg)
+            raise IndexError()
+        element_symbols = ref_ligand_df['element_symbol'].to_list()
+        if ('H' in element_symbols):
+            hydrogen_index = element_symbols.index('H')
+            subsequent_element_set = set(element_symbols[hydrogen_index:])  # A set containing elements after first hydrogen.
+            subsequent_element_set.discard('H') # Discard Hydrogen from the set.
+            if (len(subsequent_element_set) > 0):
+                exception_msg = f'{ref_name_path}: In the PDB file passed via `ref_name_path`, there should not be any hydrogen atom in the middle, i.e., the hydrogen atom(s) should be absent or at the end of the file.'
+                _LOGGER.exception(exception_msg)
+                raise ValueError()
+        else:   # If no hydrogen atom exists, pass.
+            logger_msg = f'{ref_name_path}: In the PDB file passed via `ref_name_path`, Hydrogen atoms are only present at the end of the file, check passes.'
```

Comment:
The message is kinda inconsistent as we dont have H here

Reply:
Change conducted to the comment.